### PR TITLE
OutFox improvements

### PR DESCRIPTION
**Pull request by natano**
This includes using the GaveUp() function when available.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Simply Love is compatible with current releases from the official StepMania proj
 **Compatible Releases**<br>
 ✅ [StepMania 5.0.12](https://github.com/stepmania/stepmania/releases/tag/v5.0.12)<br>
 ✅ [StepMania 5.1-b2](https://github.com/stepmania/stepmania/releases/tag/v5.1.0-b2)<br>
-✅ [OutFox 5.3 alpha 4.12.0](https://projectoutfox.com/releases/5.3.0-alpha-4.12.0) (preliminary support)<br>
```
Comment (teejusb) [inline README.md:23]: Do we mention that previous builds of OutFox (i.e. when it was still known as StepMania 5.3) are no longer supported?

File: BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 	OffCommand=function(self)
-		local fail = (GAMESTATE:GetCurMusicSeconds() < GAMESTATE:GetCurrentSong():GetLastSecond())
+		-- In OutFox and newer versions of SM 5.1 there is the GaveUp()
+		-- function available. For SM 5.0.12 and older SM 5.1 versions
+		-- we do the best we can by checking the song position. It
+		-- doesn't look like the engine broadcasts any messages that
+		-- would be helpful here.
+		local stage_stats = STATSMAN:GetCurStageStats()
+		if stage_stats.GaveUp then
+			fail = stage_stats:GaveUp()
```
Comment (teejusb) [inline BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua:13]: Should `fail` be made local? I always wonder how stuff works without the specifier.

Comment (natano) [reply BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua:13]: Oops, fixed! It's like in javascript... no specifier means it's global :cry:

File: README.md (hunk 1 above)
Comment (natano) [reply README.md:23]: updated

**Review by teejusb — APPROVED** [review]

**Pull request merged**